app/services/fees/calculator_test.go

```diff
@@ -0,0 +1,201 @@
+	mocks.MExchangeRateProvider.AssertNotCalled(t, "GetEthVsHbarRate")
+	assert.NotNil(t, err)
+	assert.Equal(t, err, InvalidTransferFee)
+	assert.False(t, valid)
+}
+
+func TestFeeCalculatorFailsWithInvalidGasPrice(t *testing.T) {
+	mocks.Setup()
+	mocks.MExchangeRateProvider.On("GetEthVsHbarRate").Return(exchangeRate, nil)
+
+	feeCalculator := NewFeeCalculator(mocks.MExchangeRateProvider, validHederaConfig())
+
+	valid, err := feeCalculator.ValidateExecutionFee(transferFee, transferAmount, invalidValue)
+	mocks.MExchangeRateProvider.AssertNotCalled(t, "GetEthVsHbarRate")
+	assert.NotNil(t, err)
+	assert.Equal(t, err, InvalidGasPrice)
+	assert.False(t, valid)
+}
+
+func TestFeeCalculatorFailsWithInvalidTransferAmount(t *testing.T) {
+	mocks.Setup()
+	mocks.MExchangeRateProvider.On("GetEthVsHbarRate").Return(exchangeRate, nil)
+
+	feeCalculator := NewFeeCalculator(mocks.MExchangeRateProvider, validHederaConfig())
+
+	valid, err := feeCalculator.ValidateExecutionFee(transferFee, invalidValue, validGasPrice)
+	mocks.MExchangeRateProvider.AssertNotCalled(t, "GetEthVsHbarRate")
+	assert.NotNil(t, err)
+	assert.Equal(t, err, InvalidTransferAmount)
+	assert.False(t, valid)
+}
+
+func TestFeeCalculatorWithInvalidRateProvider(t *testing.T) {
+	mocks.Setup()
+	mocks.MExchangeRateProvider.On("GetEthVsHbarRate").Return(float64(0), RateProviderFailure)
+
+	feeCalculator := NewFeeCalculator(mocks.MExchangeRateProvider, validHederaConfig())
+
+	valid, err := feeCalculator.ValidateExecutionFee(transferFee, transferAmount, validGasPrice)
+	mocks.MExchangeRateProvider.AssertNumberOfCalls(t, "GetEthVsHbarRate", 1)
+	assert.NotNil(t, err)
+	assert.Equal(t, err, RateProviderFailure)
+	assert.False(t, valid)
+}
+
+func TestFeeCalculatorWithManyValidators(t *testing.T) {
+	mocks.Setup()
+	mocks.MExchangeRateProvider.On("GetEthVsHbarRate").Return(exchangeRate, nil)
+	config := validHederaConfig()
+
+	feeCalculator := NewFeeCalculator(mocks.MExchangeRateProvider, config)
+
+	valid, err := feeCalculator.ValidateExecutionFee(transferFee, transferAmount, validGasPrice)
+	mocks.MExchangeRateProvider.AssertNumberOfCalls(t, "GetEthVsHbarRate", 1)
+	assert.Nil(t, err)
+	assert.True(t, valid)
+
+	config = addMoreValidatorsTo(config, 7)
+	feeCalculator = NewFeeCalculator(mocks.MExchangeRateProvider, config)
+
+	valid, err = feeCalculator.ValidateExecutionFee(transferFee, transferAmount, validGasPrice)
+	mocks.MExchangeRateProvider.AssertNumberOfCalls(t, "GetEthVsHbarRate", 2)
+	assert.NotNil(t, err)
+	assert.Equal(t, err, InsufficientFee)
+	assert.False(t, valid)
+}
+
+func TestFeeCalculatorWithZeroServiceFee(t *testing.T) {
+	mocks.Setup()
+	mocks.MExchangeRateProvider.On("GetEthVsHbarRate").Return(exchangeRate, nil)
+	config := validHederaConfig()
+	config.Client.ServiceFeePercent = 0
+
+	feeCalculator := NewFeeCalculator(mocks.MExchangeRateProvider, config)
+
+	valid, err := feeCalculator.ValidateExecutionFee(tooBigTransferFee, transferAmount, validGasPrice)
```

Comment:
Why would this fail if the service fee is 0? The fee provided is big enough (transfer of 1000, giving 1000 as fee f.e)?

I think that we should have a test:
- service fee 0
- transfer fee provided, enough to cover gas costs
- fee should be valid
- change service fee to 10%
- do `validateExecutionFee` with the same transfer fee
- fee should be invalid